org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintChangeListener.java

```diff
@@ -111,27 +109,24 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
     }
   }
 
-  private static boolean visitDelta(final Collection<IFile> changedFiles, IResourceDelta delta) {
-    IProject project = delta.getResource().getProject();
-    if (project == null) {
-      // Workspace root
-      return true;
-    }
-    if (!project.isAccessible() || !SonarLintProject.getInstance(project).isAutoEnabled()) {
+  private static boolean visitDelta(final Collection<ISonarLintFile> changedFiles, IResourceDelta delta) {
+    if (!SonarLintUtils.shouldAnalyze(delta.getResource())) {
       return false;
     }
-
-    boolean shouldAnalyzeResource = SonarLintUtils.shouldAnalyze(delta.getResource());
-    if (isChangedFile(delta) && shouldAnalyzeResource) {
-      changedFiles.add((IFile) delta.getResource());
+    ISonarLintProject sonarLintProject = (ISonarLintProject) delta.getResource().getAdapter(ISonarLintProject.class);
+    if (sonarLintProject != null) {
+      return sonarLintProject.isAutoEnabled();
+    }
+    ISonarLintFile sonarLintFile = (ISonarLintFile) delta.getResource().getAdapter(ISonarLintFile.class);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "ISonarLintFile". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1905)